cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -232,6 +232,18 @@ bool SubstraitToVeloxPlanValidator::validateScalarFunction(
   return true;
 }
 
+bool isValidArrayCast(const TypePtr& fromType, const TypePtr& toType) {
```

Comment:
Maybe, better to name it like `isSupportedArrayCast`. "valid" give me a feeling that some cast cases are acceptable in Spark semantic and some are not. But here, we just check whether it is supported in native execution.